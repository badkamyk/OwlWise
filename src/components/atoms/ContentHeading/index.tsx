import classnames from "classnames";
import { ContentTypographyType } from "./ContentTypographyType";

export default function ContentHeading({
	text,
	color = "gray",
	size = "4xl",
	className = "mb-4 tracking-tight font-extrabold",
}: ContentTypographyType) {
	return (
		<h3
			className={classnames(className, {
				"text-gray-900": color === "gray",
				"text-blue-700": color === "blue",
				"text-white": color === "white",
				"text-sm": size === "sm",
				"text-md": size === "md",
				"text-lg": size === "lg",
				"text-xl": size === "xl",
				"text-2xl": size === "2xl",
				"text-3xl": size === "3xl",
				"text-4xl": size === "4xl",
				"text-5xl": size === "5xl",
			})}
		>
			{text}
		</h3>
	);
}
