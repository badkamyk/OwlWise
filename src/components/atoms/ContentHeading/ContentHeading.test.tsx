import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import ContentHeading from "./index";

describe("ContentHeading", () => {
	it("renders with only text prop", () => {
		const result = render(<ContentHeading text="Hello World" />);
		expect(result).toBeTruthy();
		expect(screen.getByText("Hello World")).toBeTruthy();
		expect(result).toMatchSnapshot();
	});

	it("renders with all props", () => {
		const result = render(
			<ContentHeading
				text="Hello World"
				color="blue"
				size="xl"
				className="mb-4 font-extrabold tracking-tight"
			/>
		);
		expect(result).toBeTruthy();
		expect(result).toMatchSnapshot();
	});
});
