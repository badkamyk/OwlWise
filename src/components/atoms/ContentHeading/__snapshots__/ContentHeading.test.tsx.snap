// Vitest Snapshot v1

exports[`ContentHeading > renders with all props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3
        class="mb-4 font-extrabold tracking-tight text-blue-700 text-xl"
      >
        Hello World
      </h3>
    </div>
  </body>,
  "container": <div>
    <h3
      class="mb-4 font-extrabold tracking-tight text-blue-700 text-xl"
    >
      Hello World
    </h3>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ContentHeading > renders with only text prop 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3
        class="mb-4 tracking-tight font-extrabold text-gray-900 text-4xl"
      >
        Hello World
      </h3>
    </div>
  </body>,
  "container": <div>
    <h3
      class="mb-4 tracking-tight font-extrabold text-gray-900 text-4xl"
    >
      Hello World
    </h3>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
